standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -738,10 +732,13 @@ public void abortTxn(AbortTxnRequest rqst) throws NoSuchTxnException, MetaExcept
 
         if (rqst.isSetReplPolicy()) {
           sourceTxnId = rqst.getTxnid();
-          txnid = getTargetTxnId(rqst.getReplPolicy(), sourceTxnId, stmt);
-          if (txnid == -1) {
+          List<Long> targetTxnIds = getTargetTxnIdList(rqst.getReplPolicy(), Lists.newArrayList(sourceTxnId), stmt);
+          if (targetTxnIds.size() != 1) {
```

Comment:
Instead of 1, can we check for !isEmpty()? Also, shall assert if the size is 1 if not empty.

Reply:
done